int-set/src/lib.rs

```diff
@@ -576,14 +589,20 @@ where
     }
 }
 
-enum RangeIter<'a, InclusiveRangeIter, AllValuesIter>
+enum RangeIter<'a, InclusiveRangeIter, AllValuesIter, T>
 where
     InclusiveRangeIter: Iterator<Item = RangeInclusive<u32>>,
     AllValuesIter: Iterator<Item = u32>,
+    T: Domain<T>,
 {
     Inclusive {
         ranges: InclusiveRangeIter,
     },
+    InclusiveDiscontinous {
+        ranges: InclusiveRangeIter,
+        current_range: Option<RangeInclusive<u32>>,
+        phantom: PhantomData<T>,
```

Comment:
why is this necessary?

Reply:
The InclusiveDiscontinous arm needs access to the T type inside of the Iterator::next() implementation. The only way I could find to get it the T type was to make it part of RangeIter, from there the phantom data is needed to avoid an unused type parameter error.